libs/martech/analytics.js

```diff
@@ -0,0 +1,48 @@
+export function processTrackingLabels(text, charLimit = 20) {
+  return text?.trim().replace(/\s+/g, ' ').split('|').join(' ')
+    .slice(0, charLimit);
+}
+
+export function decorateDefaultLinkAnalytics(block) {
+  if (block.classList.length
+    && !block.className.includes('metadata')
+    && !block.classList.contains('link-block')
+    && !block.classList.contains('section')
+    && block.nodeName === 'DIV') {
+    let header = '';
+    let linkCount = 1;
+    block.querySelectorAll('h1, h2, h3, h4, h5, h6, a:not(.video.link-block), button, .heading-title')
+      .forEach((item) => {
+        if (item.nodeName === 'A' || item.nodeName === 'BUTTON') {
+          if (!item.hasAttribute('daa-ll')) {
+            let label = item.textContent;
+            if (label.trim() === '') {
+              label = item.getAttribute('title')
+                || item.getAttribute('aria-label')
+                || item.querySelector('img')?.getAttribute('alt')
+                || 'no label';
+            }
+            label = processTrackingLabels(label);
+            item.setAttribute('daa-ll', `${label}-${linkCount}|${header}`);
+          }
+          linkCount += 1;
+        } else {
+          header = processTrackingLabels(item.textContent);
+        }
+      });
+  }
+}
+
+export async function decorateSectionAnalytics(section, idx) {
+  document.querySelector('main')?.setAttribute('daa-im', 'true');
+  section.setAttribute('daa-lh', `s${idx + 1}`);
+  section.querySelectorAll('[data-block="true"] [data-block="true"]').forEach((block) => {
+    block.removeAttribute('data-block');
+  });
+  section.querySelectorAll('[data-block="true"]').forEach((block, blockIdx) => {
```

Comment:
Maybe these two could be grouped together. In the `section.querySelectorAll('[data-block="true"]')` items, you could check if it has a `closest` `'[data-block="true"]'` parent and remove the attribute